### PR TITLE
Add view structure and definition views to inventory client

### PR DESCRIPTION
**Pull request by Jim3Things**

### CODE REVIEW

File: simulation/internal/clients/inventory/region.go [hunk 1]
```diff
@@ -18,54 +18,98 @@ import (
...
 
-	r.itemStore.provider = r
-
 	return r, nil
 }
 
+// Definition returns the 'definition' view for this instance.  The view is
+// created on first access, and retained for the lifetime of the pdu object.
+func (r *Region) Definition() *RegionViewDefinition {
+	if r.definition == nil {
```
[the remark this answers is not in the document]
Comment (Jim3Things) [reply simulation/internal/clients/inventory/region.go:59]: It pretty much has to be already - there are too many baked in patterns where multiple calls are expected to be executed in a row (e.g. SetDetails followed by Update).

File: simulation/internal/clients/namespace/address.go [hunk 2]
```diff
@@ -428,15 +435,11 @@ func GetAddressFromKey(key string) (*Address, error) {
 
 		// We expect the first field to be one of the well know table names
 		//
-		addr.table, err = GetTableNameFromString(fields[0])
-
-		if err != nil {
-			return nil, err
-		}
+		addr.table = TableName(fields[0])
```
[the remark this answers is not in the document]
Comment (Jim3Things) [reply simulation/internal/clients/namespace/address.go:438]: Well, I've pretty much pulled all the assumptions about the table names and reduced them to 'needs to be non-blank' when creating the table name set.  Given that this code assumes a starting valid address, then it is going to have a first entry that is a table - so I just assume that it is a valid table.  Suggestions?

File: simulation/internal/clients/inventory/region.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (MikeCJohnsonUk) [reply simulation/internal/clients/inventory/region.go:59]: I thought as much. Ok.

File: simulation/internal/clients/namespace/address.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (MikeCJohnsonUk) [reply simulation/internal/clients/namespace/address.go:438]: The problem I have here is that the key string being parsed is one that has been recovered from the store via a watch event, so it's not subject to the in-memory table changes elsewhere.

The removal of the validation is only safe if the store contents (wrt key's themselves) is deemed trustworthy.

[the remark this answers is not in the document]
Comment (MikeCJohnsonUk) [reply simulation/internal/clients/namespace/address.go:438]: Following the conversation we had, let's go with this change as it is.

Ok.

**Review by MikeCJohnsonUk — APPROVED** [review]

**Pull request merged**